packages/flutter_tools/lib/src/build_system/targets/ios.dart

```diff
@@ -19,6 +19,69 @@ import 'assets.dart';
 import 'dart.dart';
 import 'icon_tree_shaker.dart';
 
+/// Copy the Flutter.framework and podspec from the engine cache.
+class UnpackIOSEngine extends Target {
+  const UnpackIOSEngine();
+
+  @override
+  List<String> get depfiles => <String>['ios_engine.d'];
+
+  @override
+  List<Target> get dependencies => const <Target>[];
+
+  @override
+  List<Source> get inputs => const <Source>[];
+
+  @override
+  String get name => 'unpack_ios_engine';
+
+  @override
+  List<Source> get outputs => <Source>[];
+
+  @override
+  Future<void> build(Environment environment) async {
+    if (environment.defines[kBuildMode] == null) {
+      throw MissingDefineException(kBuildMode, 'aot_assembly');
+    }
+    final FlutterProject flutterProject = FlutterProject.current();
+    final BuildMode buildMode = getBuildModeForName(environment.defines[kBuildMode]);
+    final String iosFramework = globals.artifacts
+      .getArtifactPath(Artifact.flutterFramework, mode: buildMode, platform: TargetPlatform.ios);
+    final String iosPodfile = globals.fs.directory(iosFramework)
+      .parent.childFile('Flutter.podspec').path;
+    final List<File> inputs = globals.fs.directory(iosFramework)
+      .listSync(recursive: true)
+      .whereType<File>()
+      .toList()..add(globals.fs.file(iosPodfile));
+    Directory outputDirectory = environment.outputDir;
+    if (flutterProject.isModule) {
+      outputDirectory = environment.outputDir.childDirectory('engine')
+        ..createSync(recursive: true);
+    }
+
+    await globals.processManager.run(<String>[
```

Comment:
Any reason not to inject globals from the start?

Reply:
The Target classes need to be `const` in the current design, so there is currently no place to inject them

Reply:
ahh, makes sense.